backend/controller/admin/local_client.go

```diff
@@ -13,8 +22,46 @@ type localClient struct {
 	*AdminService
 }
 
+type diskSchemaRetriever struct{}
+
 func newLocalClient(ctx context.Context) *localClient {
 	cm := configuration.ConfigFromContext(ctx)
 	sm := configuration.SecretsFromContext(ctx)
-	return &localClient{NewAdminService(cm, sm)}
+	return &localClient{NewAdminService(cm, sm, &diskSchemaRetriever{})}
+}
+
+func (s *diskSchemaRetriever) GetActiveSchema(ctx context.Context) (*schema.Schema, error) {
```

Comment:
Unless we compile these, I don't think we can rely on these schemas being present, which means we shouldn't use them for validation in the local case at all I suspect, as compiling before setting a secret would be quite irritating.

I can't think of a better solution right now.

Reply:
Gotcha. I thought schema.pb was always available for a module by this point, bad assumption on my part!